src/main/java/com/microsoft/sqlserver/jdbc/SQLServerSecurityUtility.java

```diff
@@ -253,25 +253,34 @@ static SqlFedAuthToken getMSIAuthToken(String resource, String msiClientId) thro
         // IMDS upgrade time can take up to 70s
         final int imdsUpgradeTimeInMs = 70 * 1000;
         final List<Integer> retrySlots = new ArrayList<>();
-        final String msiEndpoint = System.getenv("MSI_ENDPOINT");
-        final String msiSecret = System.getenv("MSI_SECRET");
 
         StringBuilder urlString = new StringBuilder();
         int retry = 1, maxRetry = 1;
 
+        // MSI_ENDPOINT and MSI_SECRET can be used instead of IDENTITY_ENDPOINT and IDENTITY_HEADER
+        String identityEndpoint = System.getenv("IDENTITY_ENDPOINT");
+        if (null == identityEndpoint) {

```

Comment:
Since this is a String, I suggest doing null == identityEndpoint || identityEndpoint .trim().isEmpty() instead.